muikku/src/main/webapp/resources/scripts/gui/announcer.js

```diff
@@ -84,6 +90,64 @@
               }
           }, this));
     },
+    _onArchiveAnnouncementsClick: function () {
+      
+      var selected = $(".an-announcements").find("input:checked");
+      var values = [];
+      var titles =[];
+      
+      $.each(selected, function(i, val){
+        var parent = $(val).parents(".an-announcement");
+        var child = parent.find(".an-announcement-topic");
+        var title = child.text();
+        
+        values.push($(val).attr("value"));
+        titles.push(title)
+      });
+      
+      
+      var formFunctions = function() {
+         var titlesContainer = $('.an-archiving-ids-container');
+         var idsInput = titlesContainer.find("input");
+         
+         
+         if(titles.length > 0){
+           $.each(titles, function(index,title){
+             titlesContainer.append("<span>" + title + "</span>");
+           });
+         }else{
+           titlesContainer.append("<span>" + getLocaleText('plugin.announcer.archiveannouncement.archiving.noneselected') + "</span>");
+           
+         }
+         idsInput.val(values);
+        
+        
+      }
+      
+      var archiveAnnouncements = function(values){
+
+         var ids = values.ids;
+         var arr = ids.split(',');
+        
+         var calls=$.map(arr, function(announcement){
+           return mApi().announcer.announcements.del(announcement);
+         });
+         
+         mApi({async: false}).batch(calls).callback(function(err, results){
```

Comment:
Probably not very big issue but using `async:false` in new code is not a good practice